### PR TITLE
Enhance/display important dates on search

### LINKED ISSUE #41075: Display only most important date when listing posts or pages
We currently display 2 dates on wordpress.com/posts/scheduled and wordpress.com/pages/scheduled

Screen Shot 2020-04-14 at 11 22 10

First one refers to the last modified date while the latter indicates when the post will be published. 

In order to reduce the amount of copy, let's remove the last modified date entirely and only display the date t's scheduled.

That way, we'll always display one date depending on post status:
- Published date for published posts.
- Scheduled date for scheduled posts.
- Last modified date for draft posts.
- Trashed date for trashed posts.

To make clear what the date means, we'd need to add appropriate labels in a similar fashion to WP Admin:

Screen Shot 2020-04-07 at 2 27 34 PM

### PR DESCRIPTION
**Pull request by millerf**
* Finishing #41242  after a rebase

#### Changes proposed in this Pull Request

| Before | After | Comment|
| ---- | ---- | ----|
| Screenshot 2020-04-18 at 19 21 32 | Screenshot 2020-04-18 at 19 21 17 |  On listing, scheduled posts/pages now display the right date (scheduled ate) |
| Screenshot 2020-04-18 at 19 23 56 |  Screenshot 2020-04-18 at 19 25 06 | Published now display "sticky" if needed | 
| Screenshot 2020-04-18 at 19 29 19  |  Screenshot 2020-04-18 at 19 26 25  | Post Search now displays the right status and appropriate date. The "sticky" status has been added. Maybe need some feedback from design? |
|  Screenshot 2020-04-18 at 19 29 05 | Screenshot 2020-04-18 at 19 31 03| Pages search now displays only one date and the right status |

Based on @sixhours remarks (https://github.com/Automattic/wp-calypso/pull/41242#pullrequestreview-396563709), some minor fixes need to be added.

Fixes #41075

### CODE REVIEW

File: client/my-sites/post-relative-time-status/index.jsx [hunk 1]
```diff
@@ -21,118 +20,139 @@ class PostRelativeTime extends React.PureComponent {
...
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
-		}
-
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		// If the content is scheduled to be release within a year, do not display the year at the end
+		const scheduledTime = scheduledDate.calendar( null, {
+			sameElse: this.props.translate( '[on] ll [at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:64]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

File: client/my-sites/post-relative-time-status/index.jsx [hunk 2]
```diff
@@ -21,118 +20,139 @@ class PostRelativeTime extends React.PureComponent {
...
-		} else if ( this.props.includeBasicStatus ) {
-			if ( status === 'draft' ) {
-				statusText = this.props.translate( 'draft' );
-			} else if ( status === 'publish' ) {
-				statusText = this.props.translate( 'published' );
-			} else if ( status === 'new' ) {
-				statusText = this.props.translate( 'Publish immediately' );
-			}
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'scheduled %(displayScheduleTime)s', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:124]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'scheduled for %(displayScheduleTime)s' )` *ES Score: 8*

**Comment by millerf** [comment]
@sixhours 

>When searching for a scheduled post, the line "Scheduled for Tomorrow at [time]" reads oddly because "tomorrow" is capitalized:
screenshot

That cannot be reproduced on this branch... It shows:
Screenshot 2020-04-20 at 22 44 11
But I think the latter format was better, as it showed the hour.

**Comment by sixhours** [comment]
I can't reproduce this behavior with this branch, either, but I can on production:

Screen Shot 2020-04-20 at 4 53 20 PM

So maybe we add the post time back, ie. "Scheduled in a day at [time]"?

**Comment by millerf** [comment]
@sixhours I just pushed a fix.

File: client/my-sites/post-relative-time-status/index.jsx [hunk 3]
```diff
@@ -21,118 +20,143 @@ class PostRelativeTime extends React.PureComponent {
...
-		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
-		}
-
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		const scheduledTime = scheduledDate.calendar( null, {
+			nextDay: this.props.translate( '[tomorrow at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:64]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **38** times:
`translate( '[Tomorrow at] LT', { context: 'momentjs format string (nextDay)'} )` *ES Score: 12*

File: client/my-sites/post-relative-time-status/index.jsx [hunk 4]
```diff
@@ -21,118 +20,143 @@ class PostRelativeTime extends React.PureComponent {
...
+				comment: 'LT refers to time (eg. 18:00)',
+			} ),
+			sameElse: this.props.translate( '[on] ll [at] LT', {
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:67]: I'm not a native English speaker, so I could be wrong, but it's "on" the right preposition to use after "scheduled"? Seems that can denote [when the post was scheduled rather than the actual schedule date](https://english.stackexchange.com/questions/72122/scheduled-on-vs-scheduled-for).

**Comment by millerf** [comment]
> I'm not a native English speaker
Me neither... I'll change "on" to "for" right away...

File: client/my-sites/post-relative-time-status/index.jsx [hunk 5]
```diff
@@ -21,118 +20,150 @@ class PostRelativeTime extends React.PureComponent {
...
-		const { status, modified, date } = post;
-		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
-		}
-
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		const sameElseForScheduled = this.props.translate( '[for] ll [at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:63]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

File: client/my-sites/post-relative-time-status/index.jsx [hunk 6]
```diff
@@ -21,118 +20,150 @@ class PostRelativeTime extends React.PureComponent {
...
+			comment:
+				'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - "at" and "for" is translated',
+		} );
+
+		const sameElse = this.props.translate( '[on] ll [at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:68]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

**Comment by mmtr** [comment]
Thanks @millerf for taking this one!

Some quick notes after testing the posts search.

Before (prod) | After (this branch)
--- | ---
Screen Shot 2020-04-21 at 11 42 35 | Screen Shot 2020-04-21 at 11 42 28

- Not sure if the comma after "Draft" or "Published" is intended but I think we can drop it.
Screen Shot 2020-04-21 at 11 48 23
Screen Shot 2020-04-21 at 11 48 11

- Published posts should display the published date rather than the last modified date
Screen Shot 2020-04-21 at 11 49 07

- Scheduled posts for tomorrow should include the "for" preposition as well.
Screen Shot 2020-04-21 at 11 49 57

- Posts pending review should display the last modified date, along with the Pending review flag. I think we can consider them as drafts, since they have not been published yet.
Screen Shot 2020-04-21 at 11 52 45

**Comment by millerf** [comment]
> Not sure if the comma after "Draft" or "Published" is intended but I think we can drop it.

It is intended, we discussed this [here](https://github.com/Automattic/wp-calypso/pull/41242#issuecomment-616680998). @sixhours thought it was misleading.

 > Scheduled posts for tomorrow should include the "for" preposition as well.

Will do

> Posts pending review 

How do you create those ones? 😅 
I remember we talked about it. I should be able blind-fix it anyway...

**Comment by mmtr** [comment]
> It is intended, we discussed this [here](https://github.com/Automattic/wp-calypso/pull/41242#issuecomment-616680998). @sixhours thought it was misleading.

Uhm, I think she was referring specifically to the published posts combined with the last modified date. If we change that to read "Published on [published_date]" as I suggested above, I don't think we need the comma.

> How do you create those ones?

By toggling the "Pending review! checkbox on the sidebar panel:

Screen Shot 2020-04-21 at 12 12 08

**Comment by millerf** [comment]
@mmtr fixed. Sorry for the delay. I have been very busy lately...

File: client/my-sites/post-relative-time-status/index.jsx [hunk 7]
```diff
@@ -21,118 +20,160 @@ class PostRelativeTime extends React.PureComponent {
...
-		if ( ! post ) {
-			// Placeholder text: "a few seconds ago" in English locale
-			return now.fromNow();
-		}
+		const isScheduledPost = this.props.post.status === 'future';
+
+		let scheduledTime;
+		if ( isScheduledPost ) {
+			scheduledTime = scheduledDate.calendar( null, {
+				nextDay: this.props.translate( '[for tomorrow at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:62]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **38** times:
`translate( '[Tomorrow at] LT', { context: 'momentjs format string (nextDay)'} )` *ES Score: 9*

File: client/my-sites/post-relative-time-status/index.jsx [hunk 8]
```diff
@@ -21,118 +20,160 @@ class PostRelativeTime extends React.PureComponent {
...
+					comment: 'LT refers to time (eg. 18:00)',
+				} ),
+				sameElse: this.props.translate( '[for] ll [at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:65]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

File: client/my-sites/post-relative-time-status/index.jsx [hunk 9]
```diff
@@ -21,118 +20,160 @@ class PostRelativeTime extends React.PureComponent {
...
+				return time.fromNow();
+			}
 
-		const { status, modified, date } = post;
-		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
+			scheduledTime = scheduledDate.calendar( null, {
+				sameElse: this.props.translate( '[on] ll [at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:77]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

**Comment by mmtr** [comment]
Thanks for the update @millerf and no need to apologize! You can take any time you need. Let me know what do you think about my notes below:

**Search: posts pending review**

Screen Shot 2020-04-23 at 15 32 41

Since posts pending reviews are actually drafts waiting for someone to review them, can we display the "Draft" status there and remove the modified date from the "pending review" flag? I think something like this would be ideal:



**Drafts list: posts pending review**

Also noted the pending review flag only appears when searching for posts. Can we display it as well when listing the drafts?

Screen Shot 2020-04-23 at 15 38 42

**Search: published posts**

There is an extra "on" we should remove.

Screen Shot 2020-04-23 at 15 41 29

**Published posts list**

Not sure we need to prefix the dates with "on". I think just the date reads ok, but don't have a strong opinion either so maybe both are fine. @Automattic/dotcom-manage-design, what's your feedback on this?

Screen Shot 2020-04-23 at 15 42 29

**Comment by millerf** [comment]
@mmtr 

I modified "pending review" to be have a "draft" status, and added "pending" status as an "extra" sticker (the same way as "sticky" is).

I removed the duplicated "on"

I am waiting on the feedback of designers for the "On" on the listing...

And "future" was moved. Not sure why I left it there in the first place. I think it was to have an exhaustive list of status, even if "published" was still missing 🤷🏻‍♂️


| Search | Listing | 
| ---- | ---- | 
| Screenshot 2020-04-24 at 19 42 49 | Screenshot 2020-04-24 at 19 42 40|

**Comment by sixhours** [comment]
I agree with @mmtr, the "On" isn't necessary. Just the date is fine. :)

**Comment by jancavan** [comment]
> the "On" isn't necessary. Just the date is fine. :)

Somehow related, I don't think "tomorrow" needs to have the first letter capitalized?

Just quickly reading the comments (so I'm not sure if this has been addressed).

**Comment by millerf** [comment]
> the "On" isn't necessary. Just the date is fine. :)

Done

> Somehow related, I don't think "tomorrow" needs to have the first letter capitalized?

That was already done.


I tried to address as much as I could. IT should be good now.

File: client/my-sites/post-relative-time-status/index.jsx [hunk 10]
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
...
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
+			let sameElseTranslation;
+			if ( noPrefix ) {
+				sameElseTranslation = this.props.translate( 'll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
+				} );
+			} else {
+				sameElseTranslation = this.props.translate( '[on] ll [at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:82]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41301)

File: client/my-sites/post-relative-time-status/index.jsx [hunk 11]
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
...
 			case 'pending':
 				return this.props.post.modified;
+
 			default:
 				return this.props.post.date;
 		}
 	}
 
-	getDisplayedTimeFromPost( post ) {
+	getDisplayedTimeForLabel( noPrefix = false ) {
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:51]: Optional: negated variables are usually hard to follow in code. Using `prefix = true` here will likely make the code a bit more easier to understand.

File: client/my-sites/post-relative-time-status/index.jsx [hunk 12]
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
...
-					? scheduledTime
-					: scheduledTime.replace( scheduledDate.format( 'Y' ), '' );
-
-			statusText = this.props.translate( 'scheduled for %(displayScheduleTime)s', {
-				comment: '%(displayScheduleTime)s is when a scheduled post is set to be published',
+		} else if ( status === 'trash' ) {
+			statusClassName += ' is-trash';
+			statusIcon = 'trash';
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'trashed %(displayScheduleTime)s', {
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:139]: @Automattic/i18n or @akirk can confirm, but I think we need to handle both status and date on the same translation. For instance, here `displayScheduleTime` contains `on [date]` for posts trashed more than a week ago, so `trashed %(displayScheduleTime)s` finally reads as `trashed on [date]`. Some languages might need to move the translated "on" part be before "trashed". To support those languages, the translatable string should be `trashed on [date]` rather than `trashed [datePrefixedWithOn]`.

The same applies for the rest of statuses (scheduled for [date], published on [date], draft last modified on [date], ...).

File: client/my-sites/post-relative-time-status/index.jsx [hunk 13]
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
...
+			} );
+		} else if ( status === 'private' ) {
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'private last modified %(displayScheduleTime)s', {
+				comment: '%(displayScheduleTime)s is when a private post or page was last modified',
+				args: {
+					displayScheduleTime,
+				},
+			} );
+		} else if ( status === 'new' ) {
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:179]: How can we test this status? Is it used for posts/pages that have been just published?

File: client/my-sites/post-relative-time-status/index.jsx [hunk 14]
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
...
+				) }
 			</span>
 		);
 	}
 
-	getStatusText() {
+	/**
+	 * Get status label
+	 *
+	 * @param {boolean} onlySticky sends back the "sticky" label. Special case as is using the same template but for is unrelated to the status
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:118]: To avoid confusion, can we maybe separate concerns and create new functions for the sticky and pending review labels?

```
getStickyLabel() {
	return (
		<span className="post-relative-time-status__status is-sticky">
			<Gridicon icon="bookmark-outline" size={ 18 } />
			<span className="post-relative-time-status__status-text">{ this.props.translate( 'sticky' ) }</span>
		</span>
	);
}

getPendingReviewLabel() {
	return (
		<span className="post-relative-time-status__status is-pending">
			<Gridicon icon="aside" size={ 18 } />
			<span className="post-relative-time-status__status-text">{ this.props.translate( 'pending review' ) }</span>
		</span>
	);
}
```

That way we can get rid of the `onlySticky`/`onlyPending` arguments here, and used the specific functions for the labels [when needed](https://github.com/Automattic/wp-calypso/blob/00c312676cd04b2a33c26e80984ed03db27fe531/client/my-sites/post-relative-time-status/index.jsx#L199-L200): 

```
{ showPublishedStatus ? this.getStatusText() : timeText }
{ post.sticky && this.getStickyLabel() }
{ post.status === 'pending' && this.getPendingReviewLabel() }
```

File: client/my-sites/post-relative-time-status/index.jsx [hunk 15]
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
...
-			}
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'scheduled %(displayScheduleTime)s', {
+				comment: '%(displayScheduleTime)s is when a scheduled post or page is set to be published',
+				args: {
+					displayScheduleTime,
+				},
+			} );
+		} else if ( status === 'draft' || status === 'pending' ) {
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:156]: Non-blocking: Since this is not a scheduled post, we could drop any schedule reference here
```suggestion
			const displayTime = this.getDisplayedTimeForLabel();
```

The same applies for the rest of statuses.

File: client/my-sites/post-relative-time-status/index.jsx [hunk 16]
```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
...
+			statusText = this.props.translate( 'published %(displayScheduleTime)s', {
+				comment: '%(displayScheduleTime)s is when a post or page was last modified',
+				args: {
+					displayScheduleTime,
+				},
+			} );
+		} else if ( status === 'private' ) {
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'private last modified %(displayScheduleTime)s', {
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:173]: Private posts should be considered as published as well. I noted we add a "Private" prefix to the title of these posts, so that should be enough: 

Screen Shot 2020-04-27 at 11 55 26

Ideally we could flag these posts using a label similar to the "Sticky"/"Pending reviews" ones and remove the title prefix, but it's fine to handle that in a follow-up.

Comment (millerf) [reply client/my-sites/post-relative-time-status/index.jsx:173]: I did create an issue with that in mind a few days ago:
https://github.com/Automattic/wp-calypso/issues/41245

I'll reference it here to follow-up

File: client/my-sites/post-relative-time-status/index.jsx (hunk 13 above)
Comment (millerf) [reply client/my-sites/post-relative-time-status/index.jsx:179]: I am not sure. I haven't been able to test it either. It is on master, though:

https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/post-relative-time-status/index.jsx#L133

File: client/my-sites/post-relative-time-status/index.jsx (hunk 12 above)
Comment (millerf) [reply client/my-sites/post-relative-time-status/index.jsx:139]: This is quite complex to tackle, as the "on" is not always present (for example : "Published yesterday" vs "Published on Mar, 23 march 2020")
And as translation texts cannot have variables to pass on the status, I would have to have to list all cases that could happen.

Also waiting to translators to confirm, but I think the "on" is the beginning of the proposition of time, so it correlates more with the date than the adjective. As in "On 23rd of March, I published, etc,..."  But I might be mistaken...

File: client/my-sites/post-relative-time-status/index.jsx (hunk 14 above)
Comment (millerf) [reply client/my-sites/post-relative-time-status/index.jsx]: I didn't want to duplicate the HTML for maintenance purposes, but I think I found a good compromise:

https://github.com/Automattic/wp-calypso/pull/41301/files#diff-d7a7d45a786a1fd8012a5e9abc0a1d78R176-R200

File: client/my-sites/post-relative-time-status/index.jsx [hunk 17]
```diff
@@ -21,121 +20,184 @@ class PostRelativeTime extends React.PureComponent {
...
-			statusClassName += ' is-sticky';
-			statusIcon = 'bookmark-outline';
-		} else if ( status === 'pending' ) {
-			statusText = this.props.translate( 'pending review' );
-			statusClassName += ' is-pending';
+		if ( status === 'trash' ) {
+			extraStatusClassName = 'is-trash';
+			statusIcon = 'trash';
+			const displayedTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'trashed %(displayedTime)s', {
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:129]: Cannot add more comments to the [previous discussion](https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415657120), so I'm copying the comments over here to continue it.

@mmtr https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415657120:

> @Automattic/i18n or @akirk can confirm, but I think we need to handle both status and date on the same translation. For instance, here `displayScheduleTime` contains `on [date]` for posts trashed more than a week ago, so `trashed %(displayScheduleTime)s` finally reads as `trashed on [date]`. Some languages might need to move the translated "on" part before "trashed" (while leaving the date after). To support those languages, the translatable string should be `trashed on [date]` rather than `trashed [datePrefixedWithOn]`.
>
> The same applies for the rest of statuses (scheduled for [date], published on [date], draft last modified on [date], ...).

@millerf https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415989089:

> This is quite complex to tackle, as the "on" is not always present (for example : "Published yesterday" vs "Published on Mar, 23 march 2020")
And as translation texts cannot have variables to pass on the status, I would have to have to list all cases that could happen.
> 
> Also waiting to translators to confirm, but I think the "on" is the beginning of the proposition of time, so it correlates more with the date than the adjective. As in "On 23rd of March, I published, etc,..."  But I might be mistaken...

Comment (mmtr) [reply client/my-sites/post-relative-time-status/index.jsx:129]: > Also waiting to translators to confirm, but I think the "on" is the beginning of the proposition of time, so it correlates more with the date than the adjective

Yeah, let's see if @Automattic/i18n or @akirk can jump in. I might be wrong as well, but languages are so different that I wouldn't assume that prepositions are always part of the date in all of them.

File: client/my-sites/post-relative-time-status/index.jsx [hunk 18]
```diff
@@ -21,121 +20,184 @@ class PostRelativeTime extends React.PureComponent {
...
+	 * Get "Pending" label
+	 */
+	getPendingLabel() {
+		return this.getLabel( this.props.translate( 'pending review' ), 'is-pending' );
+	}
+
+	/**
+	 * Get Label for the status
+	 *
+	 * @param {*} statusText text status
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:193]: Missed these jsdoc param types on my previous review. Is there any way we can make them more specific? I think they all are `string`, aren't they?

Comment (millerf) [reply client/my-sites/post-relative-time-status/index.jsx:193]: They weren't there last time :) I had to refactor part of the component...

**Comment by akirk** [comment]
I took a look at the screenshots and the code and it seems to me we still have a lot of string concats here which should be avoided. In some languages maybe the date needs to be put at the start of the phrase, this makes it impossible. Also it's hard to know if in a translation the date might depend on the rest of the phrase. Since there are not that many possible states, I'd recommend to just write them out:
- `Trashed on %(specificDate)s` (comment: Example: Trashed on March 3)
- `Trashed %(relativeTimeAgo)s` (comment: Example: Scheduled 3 days ago)
- `Published on %(specificDate)s` (comment: Example: Published on March 3)
- `Published %(relativeTimeAgo)s` (comment: Example: Published 3 days ago)
- `Last modified on %(specificDate)s` (comment: Example: Last modified on March 3)
- `Last modified %(relativeTimeAgo)s` (comment: Example: Last modified 3 days ago)
- `Scheduled %(futureRelativeDateAndTime)s` (comment: Example: Scheduled in 3 hours)
- `Scheduled for %(futureSpecificDateAndTime)s` (comment: Example: Scheduled for March 3)

Where possible, we should go for a label, like with `Pending Review`, we should do the same for `Private` and `Draft`. So for example:
- `Scheduled in 2 minutes <icon> Pending Review`
- `Scheduled for August 3 10:00 <icon> Private <icon> Sticky`
- `Last modified 3 days ago <icon> Draft <icon> Private`
- `Last modified on January 1 <icon> Draft <icon> Private`
- `Last modified on January 1 <icon> Pending Review`
- `Published 3 days ago <icon> Private <icon> Sticky`
- `Published on January 1 <icon> Private <icon> Sticky`
- `Trashed 3 days ago <icon> Private <icon> Sticky`
- `Trashed on January 1 <icon> Private <icon> Sticky`

Using a clock instead of "Last modified" and then the plain specific date or relative date is also an option. But it should only be used for "Last modified" and that date should only be displayed when the post is in the draft stage (resp. pending review).

**Comment by mmtr** [comment]
> Where possible, we should go for a label, like with `Pending Review`, we should do the same for `Private` and `Draft`.

Not sure we need a `Draft` label. It is a status like trashed, published or scheduled. In fact, the last modified date will be only displayed on drafts, which is why we want to display posts in this state as `Draft last modified on [date]`/`Draft last modified [relativeTimeAgo]`.

**Comment by millerf** [comment]
> Also it's hard to know if in a translation the date might depend on the rest of the phrase

Following our discussion about the right place of the `[on]`, please consider those examples:
```
Scheduled in 2 minutes
Scheduled for August 3 10:00
Published on January 1
```

`[in]`, `[for]` and `[on]` have the exact same syntaxic/grammatical function. So if we had to move [on] to the "published" translation block following your logic, we should also move [for] and [in] (although the last one is directly translated from the `moment` library, so it might be difficult).

I'll prepare the code for that anyway, but I think we really need some translator to help out here before doing anything 😄

**Comment by akirk** [comment]
> [in], [for] and [on] have the exact same syntaxic/grammatical function.

In English. You cannot make this assumption for other languages, some languages don't have prepositions. I only know a few but I can well see cases where a relative time needs to come at the beginning of a sentence, while an absolute time needs to be at the end. This is why we should not split sentences and join them through concat.

File: client/my-sites/post-relative-time-status/index.jsx [hunk 19]
```diff
@@ -21,121 +20,200 @@ class PostRelativeTime extends React.PureComponent {
...
-		if ( ! post ) {
-			// Placeholder text: "a few seconds ago" in English locale
-			return now.fromNow();
-		}
+		const isScheduledPost = this.props.post.status === 'future';
+
+		let displayedTime;
+		if ( isScheduledPost ) {
+			displayedTime = timestamp.calendar( null, {
+				nextDay: this.props.translate( '[tomorrow at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:61]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **38** times:
`translate( '[Tomorrow at] LT', { context: 'momentjs format string (nextDay)'} )` *ES Score: 12*

File: client/my-sites/post-relative-time-status/index.jsx [hunk 20]
```diff
@@ -21,121 +20,200 @@ class PostRelativeTime extends React.PureComponent {
...
+		switch ( status ) {
+			case 'trash':
+				if ( displayOn ) {
+					return this.props.translate( 'trashed on %(displayedTime)s', {
+						comment: '%(displayedTime)s is when a post or page was trashed',
+						args,
+					} );
+				}
+				return this.props.translate( 'trashed %(displayedTime)s', {
+					comment: '%(displayedTime)s is when a post or page was trashed',
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:124]: Given absolute/relative dates need to be at a different place for some languages, translators will need to know if this date is relative or absolute. Could you add some extra info in the comments so they can know that we use an absolute date in `trashed on/published on/scheduled for/last modified on [date]` and a relative date in `trashed/published/last modified [date]`?

**Comment by millerf** [comment]
@mmtr Thanks a lot for the deep testing. Yesterday I didn't quite finish what I wanted to do, to it is still work-in-progress. Sorry I forgot to mention it!


But anyway, I found something interesting: 
> Private posts are flagged twice now: with a prefix in the title and with a status label. Fine to address this in a follow-up.

I didn't have any private post, so I didn't see it.
It turns out it is coming from the back-end. They prefix the post title with "private: " when searching.

```
Request URL: https://public-api.wordpress.com/rest/v1.1/sites/174604573/posts?http_envelope=1&author=&number=20&order=DESC&search=&status=publish%2Cprivate&type=post&page=1
``` 
Screenshot 2020-04-29 at 18 03 44

**Comment by millerf** [comment]
>Published date is not displayed for private posts. We'd need to fix this before merging.

Fixed

**Comment by millerf** [comment]
I found another bug on `wordpress.com` that I will fix now:
You can click on a trashed page, and it sends you to a `404`...

**Comment by kwight** [comment]
> You can click on a trashed page, and it sends you to a 404... Posts cannot be clicked when trashed.

Well spotted @millerf , definitely a bug: https://github.com/Automattic/wp-calypso/issues/41640

File: client/my-sites/post-relative-time-status/index.jsx [hunk 21]
```diff
@@ -21,121 +20,208 @@ class PostRelativeTime extends React.PureComponent {
...
 			<span className="post-relative-time-status__time">
-				<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
-				<time className="post-relative-time-status__time-text" dateTime={ time }>
-					{ this.getDisplayedTimeFromPost( this.props.post ) }
-				</time>
+				{ time && (
+					<>
+						<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
+						<time className="post-relative-time-status__time-text" dateTime={ time }>
+							{ this.getDisplayedTimeForLabel( true ) }
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:98]: `getDisplayedTimeForLabel` no longer needs arguments.
```suggestion
							{ this.getDisplayedTimeForLabel() }
```

File: client/my-sites/post-relative-time-status/index.jsx [hunk 22]
```diff
@@ -21,121 +20,208 @@ class PostRelativeTime extends React.PureComponent {
...
-			return time.format( 'lll' );
+		let displayedTime;
+		if ( isScheduledPost ) {
+			displayedTime = timestamp.calendar( null, {
+				nextDay: this.props.translate( '[tomorrow at] LT', {
+					comment: 'LT refers to time (eg. 18:00)',
+				} ),
+				sameElse: this.props.translate( 'll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - and "for" is translated',
```
Comment (mmtr) [inline client/my-sites/post-relative-time-status/index.jsx:66]: Seems this refers to an old message.
```suggestion
						'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - "at" is translated',
```

**Comment by mmtr** [comment]
> It turns out it is coming from the back-end. They prefix the post title with "private: " when searching.

Nice catch! I'll file an issue so we can fix that server-side once this lands.

**Comment by millerf** [comment]
@mmtr I commited your suggestions, I think this can be landed safely now
